drafts/25-xxx-specifications.txt

```diff
@@ -99,68 +98,107 @@ li32. Fortran source fragments may be continued with a continuation
+      on a directive line are treated as a single space.
+
+ws07. Whitespace characters are significant in determining token
       boundaries for the purposes of recognizing macro names,
       in both fixed-form and free-form Fortran source fragments.
 
-ws03. Whitespace characters are significant in determining token
+ws09. Whitespace characters are significant in determining token
       boundaries for the purposes of recognizing macro names,
       in both fixed-form and free-form Fortran comment lines.
 
-ws04. In fixed-form input, macro names are not recognized as such when
+ws11. In fixed-form input, macro names are not recognized as such when
       spaces are inserted into their invocations.
 
 
-2.2 Comments
+2.4 Comments
 ------------
 
 co01. Directive lines may contain C-style '/*' ... '*/' comments.
 
-co02. Directive lines shall not contain C style '//' comments.
+co03. Directive lines shall not contain C style '//' comments.
 
-co02. '/*' ... '*/' comments on directive lines shall extend past a
+co05. '/*' ... '*/' comments on directive lines shall extend past a
       new-line only if the line ends in '\' new-line, indicating a
       continuation line. (Maybe controversial)
 
-co03. '/*' ... '*/' comments on directive lines are replaced by a
+co07. '/*' ... '*/' comments on directive lines are replaced by a
       single space, as specified in C 2023 5.1.1.2 bullet 3.
 
-co04. In a directive line, the '!' character is not interpreted as introducing
+co09. In a directive line, the '!' character is not interpreted as introducing
       a Fortran-style comment, and neither the `!` character nor any subsequent
       text are removed by the preprocessor.
 
-co05. Directive lines (by definition) cannot contain Fortran
+co11. Directive lines (by definition) cannot contain Fortran
       fixed-form 'C' or '*' style comments.
 
 
-2.4 Token lexicon
+2.5 Token lexicon
 -----------------
-[gak: List the character sequences that are recognized as separate tokens
-on directives and in Fortran source fragments. This should include
-Fortran identifiers, Fortran numeric constants, Fortran character
-constants, the Fortran operators, and the C operators used to construct
-constant-integer-expression.]
+
+The preprocessor is a token-manipulation engine (See C 2023 5.1.1.2
+Translation phases).
+As such, there is a specific lexicon of tokens recognized by
+the preprocessor (including unrecognizable tokens).
+
+We use illustrative syntax to describe the directive specifications,
+and the execution behavior of the preprocess on Fortran comment lines
+and Fortran source fragment lines.
+
+This illustrative syntax makes use of these "tokens".
+Further definition of the recognized tokens is deferred to the upcoming preprocessor syntax paper.
+
+
+to01. In definitions of object macros and function/like macros,
+      the tokens that may appear in a replacement list include those
+      allowed in Fortran and those allowed in C integer expressions.
```

Comment:
As written, this rule could easily be misread as overly constraining. 

In CPP, a `#define` replacement list may contain literally any text (other than a newline). This includes things like C keywords, or even unknown tokens (e.g. `#$&%`).

The rule is technically accurate because indeed the "tokens .. may .. include" the subset you name, but they also may include plenty of other tokens outside the named subset. IMO we should either name the complete permitted set (literally any text other than newline), or use the words "for example" to clarify we are naming a proper subset.

Reply:
I have updated the PR with more liberal wording. You are sure to have improvements.